.github/workflows/npm-publication.yml

```diff
@@ -0,0 +1,89 @@
+name: NPM Package Publication
+
+on:
+  workflow_call:
+    inputs:
+      node_version:
+        required: false
+        type: string
+        default: '16'
+      command:
+        required: false
+        type: string
+        default: 'npm publish'
+      commit:
+        required: false
+        type: boolean
+        default: false
+    secrets:
+      NPM_TOKEN:
+        required: true
+      NPM_TOTP_DEVICE:
+        required: true
+
+jobs:
+  build:
+    name: Build + Publish
+    runs-on: [ubuntu-latest]
+    environment: npm-publish
+    steps:
+      - uses: zendesk/checkout@v3
+      - name: Set up Node.js
+        uses: zendesk/setup-node@v3
+        with:
+          node-version: ${{ inputs.node_version }}
+      # Use yarn2
+      - name: Install Yarn
+        run: npm install -g yarn
+      - name: Set Yarn Version
+        run: yarn set version berry
+
+      - id: find-yarn-cache-folder
+        name: Find Yarn's cache folder
+        run: echo "::set-output name=path::$(yarn config get cacheFolder)"
```

Comment:
The `set-output` command is deprecated and will be disabled soon we should update this to the latest supported functionality

Reply:
Changed it to `echo "YARN_CACHE_FOLDER=$(yarn config get cacheFolder)" >> $GITHUB_ENV`. 
And the usage to `path: ${{ env.YARN_CACHE_FOLDER }}`